third_party/blink/renderer/bindings/core/v8/record_replay_events.cc

```diff
@@ -108,7 +108,9 @@ UserEventProbe::UserEventProbe()
 {}
 
 UserEventProbe::~UserEventProbe() {
-  ReplayNotifyAfterEvent(name_, event_target_, is_callback_);
+  if (!name_.empty()) {  // only trigger this if the probe was initialized
+    ReplayNotifyAfterEvent(name_, event_target_, is_callback_);
+  }
```

Comment:
Hmm, this code is written in a way that's hard to follow.  Calls to ReplayNotifyBeforeEvent and ReplayNotifyAfterEvent should be symmetric.  With this fix we call BeforeEvent unconditionally and AfterEvent conditionally, because BeforeEvent checks for a missing name but AfterEvent doesn't.

There are several ways to improve this but I think the different tests in BeforeEvent vs. AfterEvent are the main problem --- they led to this bug and it's hard to see at a glance whether they are equivalent.  I think it would be better if we had a ShouldNotifyEvent function which is called by both functions.  That would allow removing the gIsEventInFlight variable as then it is only used in an assertion.

Reply:
Fixed.
* Then [ran another test](https://github.com/replayio/build-test-dashboard/blob/master/272aafbd-64b9-4ac6-8e4d-f8eae21bc514) with 26 x `Passed`, 2 x `Failed`.
* Then added back the `event flag` check.